### PR TITLE
conformance: field selection

### PR DESCRIPTION
**Pull request by madi8229**
tests for field selection, it includes map fields selection and namespace resolution. Test related to protobuf does not be included.

### CODE REVIEW

File: tests/simple/testdata/fields.textproto [hunk 1]
```diff
@@ -1,2 +1,251 @@
...
+        entries {
+          key: { string_value: "k" }
+          value: { string_value: "v" }
+        }
+      }
+    }
+  }
+}
+section {
+  name: "namespace resolution"
```
Comment (JimLarson) [inline tests/simple/testdata/fields.textproto]: Call this "qualified identifier resolution" - namespaces are different.  I forgot to add the 'namespace' option to simple tests, so we can't do those yet.  Also, try to make section names short and valid as identifiers, such as "qualified_ident".  When a test fails we print out its name as `<file>/<section>/<test>`, so we want to keep those names short and without spaces.

File: tests/simple/testdata/fields.textproto [hunk 2]
```diff
@@ -1,2 +1,251 @@
...
+              key: { string_value: "name" }
+              value: { string_value: "field" }
+            }
+          }
+        }
+      }
+    }
+    value: { string_value: 'field' }
+  }
+  test {
```
[the remark this answers is not in the document]
Comment (madi8229) [reply tests/simple/testdata/fields.textproto]: done.

File: tests/simple/testdata/fields.textproto [hunk 3]
```diff
@@ -1,2 +1,251 @@
...
+    expr: '{27:false}[27]'
+    value: { bool_value: false }
+  }
+  test {
+    name: "map_value_string"
+    expr: "{'n':'x'}['n']"
+    value: { string_value: 'x' }
+  }
+  test {
+    name: "map_value_float"
```
[the remark this answers is not in the document]
Comment (madi8229) [reply tests/simple/testdata/fields.textproto]: Oops. I updated the key of the map to int64.

File: tests/simple/testdata/fields.textproto (hunk 1 above)
Comment (madi8229) [reply tests/simple/testdata/fields.textproto]: done.

**Comment by madi8229** [comment]
PTAL.

File: tests/simple/testdata/fields.textproto [hunk 4]
```diff
@@ -1,2 +1,273 @@
...
+    value: { int64_value: 5 }
+  }
+  test {
+    name: "map_key_mix_type"
+    expr: "{true:1,2:2,5u:3}[true]"
+    value: { int64_value: 1 }
+  }
+  test {
+    name: "map_value_repeat_key"
+    expr: "{true:1,false:2,true:3}[true]"
```
Comment (JimLarson) [inline tests/simple/testdata/fields.textproto]: This should fail.  Move it to broken.

Comment (madi8229) [reply tests/simple/testdata/fields.textproto]: done.

**Comment by madi8229** [comment]
PTAL.

**Pull request merged**